surefire-booter/src/main/java/org/apache/maven/surefire/booter/ForkedBooter.java

```diff
@@ -236,13 +236,17 @@ private static void exit( int returnCode, Shutdown shutdownType, CommandReader r
         switch ( shutdownType )
         {
             case KILL:
+                System.out.close();
```

Comment:
`System.in` and `System.out` are both [`PrintStream`](http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/8108ffe38ccb/src/share/classes/java/io/PrintStream.java)s. That class already is error prone like mad (autoFlush yes or no, checkError instead of throwing exceptions, etc.). It internally creates a `BufferedWriter` used for writing characters. I haven't found anything in the JDK flushing those buffers so it seems this is up to the application. I am quite sure that closing those streams manually is correct. That's what you would do in a C application as well.